tests/end2end-tests/graphstorm-ec/mgpu_test.sh

```diff
@@ -145,19 +147,21 @@ rm -fr /data/gsgnn_ec_text/*
 echo "**************dataset: Generated MovieLens EC, RGCN layer: 1, node feat: text feature, inference: full graph, exclude-training-targets: True, train_nodes 10"
 python3 $DGL_HOME/tools/launch.py --workspace $GS_HOME/training_scripts/gsgnn_ep/ --num_trainers $NUM_TRAINERS --num_servers 1 --num_samplers 0 --part_config /data/movielen_100k_ec_1p_4t_text/movie-lens-100k-text.json --ip_config ip_list.txt --ssh_port 2222 "python3 gsgnn_lm_ep.py --cf ml_lm_ec.yaml --num-gpus $NUM_TRAINERS --part-config /data/movielen_100k_ec_1p_4t_text/movie-lens-100k-text.json --exclude-training-targets True --num-classes 6 --feat-name feat --mini-batch-infer false --topk-model-to-save 1  --save-embed-path /data/gsgnn_ec_lm/emb/ --save-model-path /data/gsgnn_ec_lm/ --save-model-per-iter 1000" | tee train_log.txt
 
-error_and_exit $?
+error_and_exit ${PIPESTATUS[0]}
 
 best_epoch=$(grep "successfully save the model to" train_log.txt | tail -1 | tr -d '\n' | tail -c 1)
 echo "The best model is saved in epoch $best_epoch"
 
 echo "**************dataset: Generated MovieLens EC, node feat: text feature, do inference on saved model"
 python3 $DGL_HOME/tools/launch.py --workspace $GS_HOME/inference_scripts/ep_infer --num_trainers $NUM_INFO_TRAINERS --num_servers 1 --num_samplers 0 --part_config /data/movielen_100k_ec_1p_4t_text/movie-lens-100k-text.json --ip_config ip_list.txt --ssh_port 2222 "python3 ep_infer_lm.py --cf ml_lm_ec_infer.yaml --num-gpus $NUM_INFO_TRAINERS --part-config /data/movielen_100k_ec_1p_4t_text/movie-lens-100k-text.json --num-classes 6 --feat-name feat --mini-batch-infer false --save-embed-path /data/gsgnn_ec_lm/infer-emb/ --restore-model-path /data/gsgnn_ec_lm/epoch-$best_epoch/" | tee log.txt
 
-error_and_exit $?
+error_and_exit ${PIPESTATUS[0]}
 
 cd $GS_HOME/tests/end2end-tests/
 python3 check_infer.py --train_embout /data/gsgnn_ec_lm/emb/ --infer_embout /data/gsgnn_ec_lm/infer-emb/
 
+error_and_exit $?
```

Comment:
change to ${PIPESTATUS[0]}?

Reply:
If we do not use pipe to redirect output, we need to use `error_and_exit $?`. Otherwise, we have to use `${PIPESTATUS[0]}`.

Reply:
OK